### PR TITLE
Add AWS Service Endpoint resource

### PR DESCRIPTION
**Pull request by phillebaba**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

This change adds the resource `azuredevops_serviceendpoint_aws` that configures the Service Endpoint added by [aws-toolkit-azure-devops](https://githu

Issue Number: #48 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: website/docs/r/serviceendpoint_aws.html.markdown [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+layout: "azuredevops"
```
Comment (xuzhang3) [inline website/docs/r/serviceendpoint_aws.html.markdown:2]: This document is for AWS service connection

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_aws_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// +build all resource_serviceendpoint_aws
+// +build !exclude_serviceendpoints
```
Comment (xuzhang3) [inline azuredevops/internal/acceptancetests/resource_serviceendpoint_aws_test.go:2]: @phillebaba  Can you help add more acceptance tests to cover more scenario.  You can refer this resource [bitbucket](https://github.com/terraform-providers/terraform-provider-azuredevops/blob/master/azuredevops/internal/acceptancetests/resource_serviceendpoint_bitbucket_test.go)

File: website/docs/r/serviceendpoint_aws.html.markdown (hunk 1 above)
Comment (phillebaba) [reply website/docs/r/serviceendpoint_aws.html.markdown:2]: I must have done something wrong when moving repo, should be fixed now.

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_aws_test.go (hunk 2 above)
Comment (phillebaba) [reply azuredevops/internal/acceptancetests/resource_serviceendpoint_aws_test.go:2]: I added the same type of tests that were in the bitbucket acc test.

**Comment by phillebaba** [comment]
Hopefully my new changes will fix all of the issues.

**Review by xuzhang3 — APPROVED** [review]
Thanks @phillebaba  LGTM.

**Pull request merged**